src/mpyl/artifacts/__init__.py

```diff
@@ -40,11 +40,17 @@ def get_build_artifacts_paths(self) -> list[Path]:
         return [path for path in found_projects if path.exists()]
 
     def pull(self, branch: str) -> None:
-        if not self.artifact_repo.does_branch_exist(branch_name=branch, remote=True):
-            self.logger.info("No remote branch to pull artifacts from")
-        artifact_paths = self.get_build_artifacts_paths()
-        # pull if exists
-        # copy files to local
+        if self.artifact_repo.does_branch_exist(branch_name=branch, remote=True):
+            self.logger.info(f"Fetching branch '{branch}' from remote")
+            self.artifact_repo.fetch_branch(branch_name=branch)
+            self.artifact_repo.checkout_branch(branch_name=branch)
+
+            for artifact_path in os.listdir(self.cache_folder):
+                shutil.copytree(
+                    src=self.cache_folder / artifact_path,
+                    dst=artifact_path,
+                    dirs_exist_ok=True,
+                )
```

Comment:
I think it would be good to make a clear distinction between the gathering and copying of artifacts (from the filesystem) and the repo operations. In principle you could do everything without every using a _remote_ repo. Including unit testing.

Reply:
In theory - yes. But in practice 90% of this feature is interacting with a remote repo. If you would split it up and you would want to unit test without using a remote repo, you would basically just be testing the copying of the files from the repo to the build folder and the other way around. Which is 1 line of code, so is that worth the extra complexity unless we are sure we want to use different sources than a repo?